tests/rptest/clients/kubectl.py

```diff
@@ -142,15 +142,34 @@ def logger(self) -> Logger:
     def _local_cmd(self, cmd: list[str]):
         """Run the given command locally and return the stdout as bytes.
            Logs stdout and stderr on failure."""
+        def _prepare_output(sout: str, serr: str) -> str:
+            return f"\n--------- stdout -----------\n{sout}" \
+                   f"\n--------- stderr -----------\n{serr}\n"
+
         self._redpanda.logger.info(cmd)
-        try:
-            return subprocess.check_output(cmd, stderr=subprocess.PIPE)
-        except subprocess.CalledProcessError as e:
-            self.logger.info(
-                f'Command failed (rc={e.returncode}).\n' +
-                f'--------- stdout -----------\n{e.stdout.decode()}' +
-                f'--------- stderr -----------\n{e.stderr.decode()}')
-            raise
+
+        # Using text mode to get strings, not binary
+        # Following code will capture all output to log and work
+        # significantly faster than 'check_output'
+        process = subprocess.Popen(cmd,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   text=True)
+        s_out, s_err = process.communicate()
+        self.logger.debug(_prepare_output(s_out, s_err))
+
+        # TODO: Handle s_err output
+        # It will be hard to detect errors as a lot of apps and utils
+        # just uses stderr as normal output. For example, tsh
+
+        process.kill()
```

Comment:
You don't need `kill` here, I don't think?

Reply:
The process class in memory will not be freed in full otherwise. Also, there might be rare situations when RC exists and the process in still alive. Instead of p.join() I use kill to make sure that there will be no process hang of any kind.

Reply:
Also, this probably needs update to handle timeouts and it will justify use of kill(). As process is not killed on timeout.

Reply:
Timeout is a good reason to kill the process, but as you point out we won't `kill` it here because that's an exception and will bypass this.

I don't think there is any need to call kill() here. Can you provide any reference? The docs certainly don't agree: the require `communciate` to wait for process termination before returning (unless timeout). I checked the CPython code and kill does not clean up the process class, it simply sends a kill signal to the process, and if the return code is available (which it will be after communicate by construction) it doesn't do [anything at all](https://github.com/python/cpython/blob/main/Lib/subprocess.py#L1663).